src/kubernetes/resource_version.rs

```diff
@@ -0,0 +1,92 @@
+//! Shared state bits for watch implementations.
+
+use k8s_openapi::apimachinery::pkg::apis::meta::v1::{ObjectMeta, WatchEvent};
+use k8s_openapi::Metadata;
+use std::ops::Deref;
+
+/// Resource version state in the context of a chain of watch requests.
+#[derive(Debug, Clone)]
+pub struct State(Option<String>);
+
+impl State {
+    /// Create a new resource version [`State`].
+    pub fn new() -> Self {
+        Self(None)
+    }
+
+    /// Update the resource version from a candidate obtained earlier.
+    pub fn update(&mut self, candidate: Candidate) {
```

Comment:
Since you have a `get` and an `update` it might make sense for the `update` to return the old value instead of immediately dropping.